pkg/util/blob/blob.go

```diff
@@ -0,0 +1,153 @@
+package blob
```

Comment:
This file is joining two things together which I'd rather keep separate.
1) The pkg/cluster-specific code by which we get access to cluster blobs (NewBlobServiceForCluster)
2) Modification of the blob storage API for supposed better usability/testability.

WRT 1), I think it makes sense to keep this code in pkg/cluster rather than move it to pkg/util.

WRT 2), if/when we add egress lockdown code it is likely we will want to use storage blobs and queues more.  For that reason I would rather avoid adding smarts to the util storage API but rather keep very true to the original like we've done for ARM clients in pkg/util/azureclient up to now.  It will make it easier to extend the wrapper.  PTAL at https://github.com/openshift/openshift-azure/blob/master/pkg/util/azureclient/storage/storage.go (excluding the retry code which should probably not have gone in there) for an example of the approach I'd prefer to take if possible.

Following 2), there should be fewer/no changes to pkg/cluster/graph.go, etc., for example, as we're not changing the native API.